content/storage-history/history.textile

```diff
@@ -0,0 +1,633 @@
+    ```
+
+    ```[csharp]
+    AblyRest rest = new AblyRest("{{API_KEY}}");
+    IRestChannel channel = rest.Channels.Get("{{RANDOM_CHANNEL_NAME}}");
+    await channel.PublishAsync("example", "message data");
+    PaginatedResult<Message> resultPage = await channel.HistoryAsync();
+    Message recentMessage = resultPage.Items[0];
+    Console.WriteLine("Most recent message: " + recentMessage.Id + " - " + recentMessage.Data);
+    ```
+
+    ```[objc]
+    ARTRest *rest = [[ARTRest alloc] initWithKey:@"{{API_KEY}}"];
+    ARTRestChannel *channel = [rest.channels get:@"{{RANDOM_CHANNEL_NAME}}"];
+    [channel publish:@"example" data:@"message data"];
+    [channel history:^(ARTPaginatedResult<ARTMessage *> *resultPage, ARTErrorInfo *error) {
+        ARTMessage *recentMessage = resultPage.items[0];
+        NSLog(@"Most recent message: %@ - %@", recentMessage.id, recentMessage.data);
+    }];
+    ```
+
+    ```[swift]
+    let rest = ARTRest(key: "{{API_KEY}}")
+    let channel = rest.channels.get("{{RANDOM_CHANNEL_NAME}}")
+    channel.publish("example", data: "message data")
+    channel.history { resultPage, error in
+        let recentMessage = resultPage!.items[0] as! ARTMessage
+        print("Most recent message: \(recentMessage.id) - \(recentMessage.data)")
+    }
+    ```
+
+    ```[go]
+    rest, err := ably.NewRestClient(ably.NewClientOptions("{{API_KEY}}"))
+    channel := rest.Channels.Get("{{RANDOM_CHANNEL_NAME}}")
+    err = channel.Publish("example", "message data")
+    page, err := channel.History(nil)
+    fmt.Println("Last published message: %s\n", page.Messages[0].Data)
+
+On success, the returned @PaginatedResult@ encapsulates an array of Message objects corresponding to the current page of results. @PaginatedResult@ supports pagination using next and first methods.
+
+Failure to retrieve the message history will raise an @AblyException@.
+
+h4(#channel-parameters). Channel history parameters
+
+There is only one parameter for @history@: 
+
+- options := ensures that clients disconnected abruptly from Ably can resume their connection
+
+h4(#channel-option-parameters). Option parameters
+
+These are the parameters you can add to your @options@ object, note that @untilAttach@ is only available for Realtime:
+
+|_. Parameter |_. Description |
+| start | beginning of time earliest time in milliseconds since the epoch for any messages retrieved |
+| end | current time latest time in milliseconds since the epoch for any messages retrieved |
+| direction | backwards forwards or backwards |
+| limit | 100 maximum number of messages to retrieve up to 1,000 |
+| untilAttach | false when true, ensures message history is up until the point of the channel being attached. See "continuous history":#continuous-history for more info. Requires the direction to be backwards (the default). If the Channel is not attached, or if direction is set to forwards, this option will result in an error. Realtime Only. |
```

Comment:
"false when true"? A few instances of properties that should be in @ and some odd capitalisation too.